runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java

```diff
@@ -950,8 +1002,10 @@ public DataflowPipelineJob run(Pipeline pipeline) {
             .addAllCapabilities(Environments.getJavaCapabilities())
             .build());
 
-    RunnerApi.Pipeline portablePipelineProto =
+    RunnerApi.Pipeline pipelineProto =
         PipelineTranslation.toProto(pipeline, portableComponents, false);
+    List<DataflowPackage> packages = stageArtifacts(pipelineProto);
+    RunnerApi.Pipeline portablePipelineProto = resolveArtifacts(pipelineProto);
```

Comment:
We should use a single variable instead of maintaining multiple variables for the pipeline for here and below (pipelineProto and portablePipelineProto)

Reply:
Done.